### PR TITLE
[OpenSSL] refactor openssl ports.

### PR DESCRIPTION
**Pull request by Neumann-A**
- remove platform dependent `openssl-<platform>` ports
-> moved into a subdir in the openssl port itself
- removed unused patches on windows
- fixed wrong version in the main openssl port
- added a wrapper to the windows port for static libraries (kind of tired fixing https://gitlab.kitware.com/cmake/cmake/-/issues/19263 in downstream ports; to be explicit qt6-base...)
- download and conflict check in main portfile
- usage installation in main portfile
- deprecated other `openssl-<platform>` (not removed due to `vcpkg upgrade`)

Benefits:
- `vcpkg remove openssl` actually removes openssl
- bit less code duplication across the ports
- only 1 version field instead of 4. 

Drawbacks:
- CI has to rebuild for all platforms on changes.
- probably more cleanup/code reduction possible

Future ToDo:
- Remove extra UWP folder. Windows/UWP seems to be doing a lot of the same things.

### CODE REVIEW

**Comment by mcmtroffaes** [comment]
Oh boy, what a coincidence! I'll put #14309 on hold.

File: ports/openssl-uwp/CONTROL [hunk 1]
```diff
@@ -1,5 +1,4 @@
 Source: openssl-uwp
 Version: 1.1.1h
-Port-Version: 1
-Description: OpenSSL is an open source project that provides a robust, commercial-grade, and full-featured toolkit for the Transport Layer Security (TLS) and Secure Sockets Layer (SSL) protocols. It is also a general-purpose cryptography library.
+Description: depreacted OpenSSL port
```
Comment (mcmtroffaes) [inline ports/openssl-uwp/CONTROL:3]:
```suggestion
Description: Deprecated OpenSSL port
```

File: ports/openssl-windows/CONTROL [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Source: openssl-windows
 Version: 1.1.1h
-Description: OpenSSL is an open source project that provides a robust, commercial-grade, and full-featured toolkit for the Transport Layer Security (TLS) and Secure Sockets Layer (SSL) protocols. It is also a general-purpose cryptography library.
+Description: deprecated OpenSSL port
```
Comment (BillyONeal) [inline ports/openssl-windows/CONTROL:3]:
```suggestion
Description: Deprecated OpenSSL port
```

File: ports/openssl/unix/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,48 @@
+if (NOT VCPKG_TARGET_IS_MINGW)
+    vcpkg_fail_port_install(MESSAGE "${PORT} is only for openssl on Unix-like systems" ON_TARGET "UWP" "Windows")
+endif()
+
+vcpkg_extract_source_archive_ex(
+    OUT_SOURCE_PATH MASTER_COPY_SOURCE_PATH
+    ARCHIVE ${ARCHIVE}
```
Comment (BillyONeal) [inline ports/openssl/unix/portfile.cmake:7]:
```suggestion
    ARCHIVE "${ARCHIVE}"
```

File: ports/openssl/uwp/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        LOGNAME configure-perl-${TARGET_TRIPLET}-${VCPKG_BUILD_TYPE}-rel
+    )
+    message(STATUS "Configure ${TARGET_TRIPLET}-rel done")
+
+    message(STATUS "Build ${TARGET_TRIPLET}-rel")
+    # Openssl's buildsystem has a race condition which will cause JOM to fail at some point.
+    # This is ok; we just do as much work as we can in parallel first, then follow up with a single-threaded build.
+    make_directory(${SOURCE_PATH_RELEASE}/inc32/openssl)
+    execute_process(
+        COMMAND ${JOM} -k -j $ENV{NUMBER_OF_PROCESSORS} -f ${OPENSSL_MAKEFILE} build_libs
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:75]:
```suggestion
        COMMAND "${JOM}" -k -j ${VCPKG_CONCURRENCY} -f "${OPENSSL_MAKEFILE}" build_libs
```

File: ports/openssl/windows/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    no-tests
+    -utf-8
+    ${OPENSSL_SHARED}
+)
+
+if(DEFINED OPENSSL_USE_NOPINSHARED)
+    set(CONFIGURE_OPTIONS ${CONFIGURE_OPTIONS} no-pinshared)
+endif()
+
+set(CONFIGURE_COMMAND ${PERL} Configure ${CONFIGURE_OPTIONS})
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:32]:
```suggestion
set(CONFIGURE_COMMAND "${PERL}" Configure ${CONFIGURE_OPTIONS})
```

File: ports/openssl/windows/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+        COMMAND ${CONFIGURE_COMMAND} debug-${OPENSSL_ARCH} "--prefix=${OPENSSLDIR_DEBUG}" "--openssldir=${OPENSSLDIR_DEBUG}" -FS
+        WORKING_DIRECTORY ${SOURCE_PATH_DEBUG}
+        LOGNAME configure-perl-${TARGET_TRIPLET}-${VCPKG_BUILD_TYPE}-dbg
+    )
+    message(STATUS "Configure ${TARGET_TRIPLET}-dbg done")
+
+    message(STATUS "Build ${TARGET_TRIPLET}-dbg")
+    make_directory(${SOURCE_PATH_DEBUG}/inc32/openssl)
+    execute_process(
+        COMMAND ${JOM} -k -j $ENV{NUMBER_OF_PROCESSORS} -f ${OPENSSL_MAKEFILE}
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:113]:
```suggestion
        COMMAND "${JOM}" -k -j ${VCPKG_CONCURRENCY} -f "${OPENSSL_MAKEFILE}"
```

File: ports/openssl/windows/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        WORKING_DIRECTORY ${SOURCE_PATH_DEBUG}
+        OUTPUT_FILE ${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-dbg-0-out.log
+        ERROR_FILE ${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-dbg-0-err.log
+    )
+    vcpkg_execute_required_process(
+        COMMAND nmake -f ${OPENSSL_MAKEFILE} install_sw install_ssldirs
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:119]:
```suggestion
        COMMAND nmake -f "${OPENSSL_MAKEFILE}" install_sw install_ssldirs
```

File: ports/openssl/windows/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    message(STATUS "Build ${TARGET_TRIPLET}-dbg done")
+endif()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/certs)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/private)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/engines-1_1)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/certs)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/engines-1_1)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/private)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:132]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/certs")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/private")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/engines-1_1")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/certs")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/engines-1_1")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/private")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

File: ports/openssl/portfile.cmake [hunk 9]
```diff
@@ -1,2 +1,25 @@
...
+if(VCPKG_TARGET_IS_UWP)
+    include("${CMAKE_CURRENT_LIST_DIR}/uwp/portfile.cmake")
+elseif(VCPKG_TARGET_IS_WINDOWS)
+    include("${CMAKE_CURRENT_LIST_DIR}/windows/portfile.cmake")
+else()
+    include("${CMAKE_CURRENT_LIST_DIR}/unix/portfile.cmake")
+endif()
+
+
 file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/openssl/)
```
Comment (BillyONeal) [inline ports/openssl/portfile.cmake:25]:
```suggestion
file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/")
```

File: ports/openssl/uwp/portfile.cmake [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+set(OPENSSL_MAKEFILE "makefile")
+
+file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg)
+
+
+if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
+
+    # Copy openssl sources.
+    message(STATUS "Copying openssl release source files...")
+    file(GLOB OPENSSL_SOURCE_FILES ${SOURCE_PATH}/*)
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:53]:
```suggestion
    file(GLOB OPENSSL_SOURCE_FILES "${SOURCE_PATH}/*")
```

File: ports/openssl/uwp/portfile.cmake [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+    foreach(SOURCE_FILE ${OPENSSL_SOURCE_FILES})
+        file(COPY ${SOURCE_FILE} DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel")
+    endforeach()
+    message(STATUS "Copying openssl release source files... done")
+    set(SOURCE_PATH_RELEASE "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel")
+
+    set(OPENSSLDIR_RELEASE ${CURRENT_PACKAGES_DIR})
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:60]:
```suggestion
    set(OPENSSLDIR_RELEASE "${CURRENT_PACKAGES_DIR}")
```

File: ports/openssl/uwp/portfile.cmake [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+
+    message(STATUS "Configure ${TARGET_TRIPLET}-rel")
+    vcpkg_execute_required_process(
+        COMMAND ${CONFIGURE_COMMAND} ${OPENSSL_ARCH} "--prefix=${OPENSSLDIR_RELEASE}" "--openssldir=${OPENSSLDIR_RELEASE}" -FS
+        WORKING_DIRECTORY ${SOURCE_PATH_RELEASE}
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:65]:
```suggestion
        WORKING_DIRECTORY "${SOURCE_PATH_RELEASE}"
```

File: ports/openssl/uwp/portfile.cmake [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+
+    message(STATUS "Build ${TARGET_TRIPLET}-rel")
+    # Openssl's buildsystem has a race condition which will cause JOM to fail at some point.
+    # This is ok; we just do as much work as we can in parallel first, then follow up with a single-threaded build.
+    make_directory(${SOURCE_PATH_RELEASE}/inc32/openssl)
+    execute_process(
+        COMMAND ${JOM} -k -j $ENV{NUMBER_OF_PROCESSORS} -f ${OPENSSL_MAKEFILE} build_libs
+        WORKING_DIRECTORY ${SOURCE_PATH_RELEASE}
+        OUTPUT_FILE ${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-rel-0-out.log
+        ERROR_FILE ${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-rel-0-err.log
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:78]:
```suggestion
        WORKING_DIRECTORY "${SOURCE_PATH_RELEASE}"
        OUTPUT_FILE "${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-rel-0-out.log"
        ERROR_FILE "${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-rel-0-err.log"
```

File: ports/openssl/uwp/portfile.cmake [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+    )
+    vcpkg_execute_required_process(
+        COMMAND nmake -f ${OPENSSL_MAKEFILE} install_dev
+        WORKING_DIRECTORY ${SOURCE_PATH_RELEASE}
+        LOGNAME build-${TARGET_TRIPLET}-rel-1)
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:83]:
```suggestion
        COMMAND nmake -f "${OPENSSL_MAKEFILE}" install_dev
        WORKING_DIRECTORY "${SOURCE_PATH_RELEASE}"
        LOGNAME build-${TARGET_TRIPLET}-rel-1)
```

File: ports/openssl/uwp/portfile.cmake [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+    message(STATUS "Build ${TARGET_TRIPLET}-rel done")
+endif()
+
+
+if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
+    # Copy openssl sources.
+    message(STATUS "Copying openssl debug source files...")
+    file(GLOB OPENSSL_SOURCE_FILES ${SOURCE_PATH}/*)
+    foreach(SOURCE_FILE ${OPENSSL_SOURCE_FILES})
+        file(COPY ${SOURCE_FILE} DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg")
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:94]:
```suggestion
        file(COPY "${SOURCE_FILE}" DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg")
```

File: ports/openssl/uwp/portfile.cmake [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+    message(STATUS "Copying openssl debug source files... done")
+    set(SOURCE_PATH_DEBUG "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg")
+
+    set(OPENSSLDIR_DEBUG ${CURRENT_PACKAGES_DIR}/debug)
+
+    message(STATUS "Configure ${TARGET_TRIPLET}-dbg")
+    vcpkg_execute_required_process(
+        COMMAND ${CONFIGURE_COMMAND} debug-${OPENSSL_ARCH} "--prefix=${OPENSSLDIR_DEBUG}" "--openssldir=${OPENSSLDIR_DEBUG}" -FS
+        WORKING_DIRECTORY ${SOURCE_PATH_DEBUG}
+        LOGNAME configure-perl-${TARGET_TRIPLET}-${VCPKG_BUILD_TYPE}-dbg
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:105]:
```suggestion
        WORKING_DIRECTORY "${SOURCE_PATH_DEBUG}"
        LOGNAME configure-perl-${TARGET_TRIPLET}-${VCPKG_BUILD_TYPE}-dbg
```

File: ports/openssl/uwp/portfile.cmake [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+    # Copy openssl sources.
+    message(STATUS "Copying openssl debug source files...")
+    file(GLOB OPENSSL_SOURCE_FILES ${SOURCE_PATH}/*)
+    foreach(SOURCE_FILE ${OPENSSL_SOURCE_FILES})
+        file(COPY ${SOURCE_FILE} DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg")
+    endforeach()
+    message(STATUS "Copying openssl debug source files... done")
+    set(SOURCE_PATH_DEBUG "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg")
+
+    set(OPENSSLDIR_DEBUG ${CURRENT_PACKAGES_DIR}/debug)
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:99]:
```suggestion
    set(OPENSSLDIR_DEBUG "${CURRENT_PACKAGES_DIR}/debug")
```

File: ports/openssl/uwp/portfile.cmake [hunk 18]
```diff
@@ -0,0 +1,156 @@
...
+    execute_process(
+        COMMAND ${JOM} -k -j $ENV{NUMBER_OF_PROCESSORS} -f ${OPENSSL_MAKEFILE} build_libs
+        WORKING_DIRECTORY ${SOURCE_PATH_DEBUG}
+        OUTPUT_FILE ${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-dbg-0-out.log
+        ERROR_FILE ${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-dbg-0-err.log
+    )
+    vcpkg_execute_required_process(
+        COMMAND nmake -f ${OPENSSL_MAKEFILE} install_dev
+        WORKING_DIRECTORY ${SOURCE_PATH_DEBUG}
+        LOGNAME build-${TARGET_TRIPLET}-dbg-1)
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:120]:
```suggestion
    make_directory("${SOURCE_PATH_DEBUG}/inc32/openssl")
    execute_process(
        COMMAND "${JOM}" -k -j ${VCPKG_CONCURRENCY} -f "${OPENSSL_MAKEFILE}" build_libs
        WORKING_DIRECTORY "${SOURCE_PATH_DEBUG}"
        OUTPUT_FILE "${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-dbg-0-out.log"
        ERROR_FILE "${CURRENT_BUILDTREES_DIR}/build-${TARGET_TRIPLET}-dbg-0-err.log"
    )
    vcpkg_execute_required_process(
        COMMAND nmake -f "${OPENSSL_MAKEFILE}" install_dev
        WORKING_DIRECTORY "${SOURCE_PATH_DEBUG}"
        LOGNAME build-${TARGET_TRIPLET}-dbg-1)
```

File: ports/openssl/uwp/portfile.cmake [hunk 19]
```diff
@@ -0,0 +1,156 @@
...
+    ${CURRENT_PACKAGES_DIR}/debug/bin/openssl.exe
+    ${CURRENT_PACKAGES_DIR}/debug/openssl.cnf
+    ${CURRENT_PACKAGES_DIR}/openssl.cnf
+    ${CURRENT_PACKAGES_DIR}/ct_log_list.cnf
+    ${CURRENT_PACKAGES_DIR}/ct_log_list.cnf.dist
+    ${CURRENT_PACKAGES_DIR}/openssl.cnf.dist
+    ${CURRENT_PACKAGES_DIR}/debug/ct_log_list.cnf
+    ${CURRENT_PACKAGES_DIR}/debug/ct_log_list.cnf.dist
+    ${CURRENT_PACKAGES_DIR}/debug/openssl.cnf.dist
+)
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:144]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/certs")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/private")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/engines-1_1")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/certs")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/engines-1_1")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/private")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")

file(REMOVE
    "${CURRENT_PACKAGES_DIR}/bin/openssl.exe"
    "${CURRENT_PACKAGES_DIR}/debug/bin/openssl.exe"
    "${CURRENT_PACKAGES_DIR}/debug/openssl.cnf"
    "${CURRENT_PACKAGES_DIR}/openssl.cnf"
    "${CURRENT_PACKAGES_DIR}/ct_log_list.cnf"
    "${CURRENT_PACKAGES_DIR}/ct_log_list.cnf.dist"
    "${CURRENT_PACKAGES_DIR}/openssl.cnf.dist"
    "${CURRENT_PACKAGES_DIR}/debug/ct_log_list.cnf"
    "${CURRENT_PACKAGES_DIR}/debug/ct_log_list.cnf.dist"
    "${CURRENT_PACKAGES_DIR}/debug/openssl.cnf.dist"
)
```

File: ports/openssl/uwp/portfile.cmake [hunk 20]
```diff
@@ -0,0 +1,156 @@
...
+string(REPLACE "<winsock.h>" "<winsock2.h>" _contents "${_contents}")
+file(WRITE "${CURRENT_PACKAGES_DIR}/include/openssl/dtls1.h" "${_contents}")
+
+file(READ "${CURRENT_PACKAGES_DIR}/include/openssl/rand.h" _contents)
+string(REPLACE "#  include <windows.h>" "#ifndef _WINSOCKAPI_\n#define _WINSOCKAPI_\n#endif\n#  include <windows.h>" _contents "${_contents}")
+file(WRITE "${CURRENT_PACKAGES_DIR}/include/openssl/rand.h" "${_contents}")
+
+vcpkg_copy_pdbs()
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (BillyONeal) [inline ports/openssl/uwp/portfile.cmake:156]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

Yes I know we are inconsistent about this one :(

File: ports/openssl/windows/portfile.cmake [hunk 21]
```diff
@@ -0,0 +1,172 @@
+vcpkg_fail_port_install(MESSAGE "${PORT} is only for Windows Desktop" ON_TARGET "UWP" "Linux" "OSX")
+
+vcpkg_extract_source_archive_ex(
+    OUT_SOURCE_PATH SOURCE_PATH
+    ARCHIVE ${ARCHIVE}
+)
+
+vcpkg_find_acquire_program(NASM)
+get_filename_component(NASM_EXE_PATH ${NASM} DIRECTORY)
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:9]:
```suggestion
get_filename_component(NASM_EXE_PATH "${NASM}" DIRECTORY)
```

File: ports/openssl/windows/portfile.cmake [hunk 22]
```diff
@@ -0,0 +1,172 @@
...
+    set(OPENSSL_ARCH VC-WIN32-ARM)
+elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+    set(OPENSSL_ARCH VC-WIN64-ARM)
+else()
+    message(FATAL_ERROR "Unsupported target architecture: ${VCPKG_TARGET_ARCHITECTURE}")
+endif()
+
+set(OPENSSL_MAKEFILE "makefile")
+
+file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg)
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:48]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel"
                    "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg")
```

File: ports/openssl/windows/portfile.cmake [hunk 23]
```diff
@@ -0,0 +1,172 @@
...
+
+file(REMOVE
+    ${CURRENT_PACKAGES_DIR}/ct_log_list.cnf
+    ${CURRENT_PACKAGES_DIR}/ct_log_list.cnf.dist
+    ${CURRENT_PACKAGES_DIR}/openssl.cnf.dist
+    ${CURRENT_PACKAGES_DIR}/debug/bin/openssl.exe
+    ${CURRENT_PACKAGES_DIR}/debug/ct_log_list.cnf
+    ${CURRENT_PACKAGES_DIR}/debug/ct_log_list.cnf.dist
+    ${CURRENT_PACKAGES_DIR}/debug/openssl.cnf
+    ${CURRENT_PACKAGES_DIR}/debug/openssl.cnf.dist
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:142]:
```suggestion
    "${CURRENT_PACKAGES_DIR}/ct_log_list.cnf"
    "${CURRENT_PACKAGES_DIR}/ct_log_list.cnf.dist"
    "${CURRENT_PACKAGES_DIR}/openssl.cnf.dist"
    "${CURRENT_PACKAGES_DIR}/debug/bin/openssl.exe"
    "${CURRENT_PACKAGES_DIR}/debug/ct_log_list.cnf"
    "${CURRENT_PACKAGES_DIR}/debug/ct_log_list.cnf.dist"
    "${CURRENT_PACKAGES_DIR}/debug/openssl.cnf"
    "${CURRENT_PACKAGES_DIR}/debug/openssl.cnf.dist"
```

File: ports/openssl/windows/portfile.cmake [hunk 24]
```diff
@@ -0,0 +1,172 @@
...
+)
+
+file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools/openssl/)
+file(RENAME ${CURRENT_PACKAGES_DIR}/bin/openssl.exe ${CURRENT_PACKAGES_DIR}/tools/openssl/openssl.exe)
+file(RENAME ${CURRENT_PACKAGES_DIR}/openssl.cnf ${CURRENT_PACKAGES_DIR}/tools/openssl/openssl.cnf)
+
+vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/openssl)
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:149]:
```suggestion
file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/openssl/")
file(RENAME "${CURRENT_PACKAGES_DIR}/bin/openssl.exe" "${CURRENT_PACKAGES_DIR}/tools/openssl/openssl.exe")
file(RENAME "${CURRENT_PACKAGES_DIR}/openssl.cnf" "${CURRENT_PACKAGES_DIR}/tools/openssl/openssl.cnf")

vcpkg_copy_tool_dependencies("${CURRENT_PACKAGES_DIR}/tools/openssl")
```

File: ports/openssl/windows/portfile.cmake [hunk 25]
```diff
@@ -0,0 +1,172 @@
...
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    # They should be empty, only the exes deleted above were in these directories
+    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/bin/)
+    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin/)
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:154]:
```suggestion
    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin/")
    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin/")
```

File: ports/openssl/windows/portfile.cmake [hunk 26]
```diff
@@ -0,0 +1,172 @@
...
+    "#  include <windows.h>"
+    "#ifndef _WINSOCKAPI_\n#define _WINSOCKAPI_\n#endif\n#  include <windows.h>"
+)
+
+vcpkg_copy_pdbs()
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file(COPY ${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/openssl)
+endif()
```
Comment (BillyONeal) [inline ports/openssl/windows/portfile.cmake:172]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
    file(COPY "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/openssl")
endif()
```

**Review by BillyONeal — APPROVED** [review]
I almost certainly missed some quotes but :)

**Comment by JackBoosY** [comment]
This is indeed another good way to solve the same port in different triplets.

**Comment by Neumann-A** [comment]
@JackBoosY: It should be the only allowed way ;)

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
@Neumann-A Thanks for your help!

**Pull request merged**